### PR TITLE
Quick fix to remove sharedPreferences call/legacy migration

### PR DESCRIPTION
**Pull request by dantetam**
Will handle fixing the tests later. There are too many tests to fix right now and I want to unblock the customer.

### CODE REVIEW

**Review by haoliu-amp — COMMENTED** [review]
LGTM. @djih how do you feel?

File: src/main/java/com/amplitude/api/AmplitudeClient.java [hunk 1]
```diff
@@ -334,11 +334,6 @@ public synchronized AmplitudeClient initializeInternal(
             if (!initialized) {
                 // this try block is idempotent, so it's safe to retry initialize if failed
                 try {
-                    if (instanceName.equals(Constants.DEFAULT_INSTANCE)) {
-                        AmplitudeClient.upgradePrefs(context);
```
Comment (djih) [inline src/main/java/com/amplitude/api/AmplitudeClient.java:338]: do you want to clean up these methods too? `upgradePrefs` and `upgradeSharedPrefsToDB`

**Review by djih — APPROVED** [review]

**Comment by djih** [comment]
@haoliu-amp  I think it's fine as long as we're okay breaking backwards compatibility from super legacy versions of Android SDK that are still using sharedPrefs (probably no one is using these versions anymore)

File: src/main/java/com/amplitude/api/AmplitudeClient.java (hunk 1 above)
Comment (dantetam) [reply src/main/java/com/amplitude/api/AmplitudeClient.java:338]: Going to clean up everything, including the tests @djih

Comment (dantetam) [reply src/main/java/com/amplitude/api/AmplitudeClient.java:338]: In a different branch and PR

**Pull request merged**